src/compas_fab/backends/vrep/backend_features/vrep_inverse_kinematics.py

```diff
@@ -59,5 +59,7 @@ def inverse_kinematics(self, robot, frame_WCF, start_configuration=None, group=N
 
         states = self.client.find_raw_robot_states(group, frame_to_vrep_pose(frame_WCF, self.client.scale), gantry_joint_limits, arm_joint_limits, max_trials, max_results)
 
-        return [config_from_vrep(states[i:i + num_joints], self.client.scale)
-                for i in range(0, len(states), num_joints)]
+        joint_names = robot.get_configurable_joint_names()
+
+        for i in range(0, len(states), num_joints):
+            yield config_from_vrep(states[i: i+num_joints], self.client.scale).joint_values, joint_names
```

Comment:
This is cool :+1: I think now, besides the fact that Vrep has no sematics (and that messes up with the group selection), it would almost be possible to call IK from the Robot class and have it properly forwarded into V-REP.